10up/wp-local-docker-v2
# Feature/use new snapshots

### Description of the Change

This PR updates the local docker system to utilise the new Snapshots WP-CLI command rather than the prior docker-based WP Snapshots system.

This has the following benefits:

- Removes the need for a dedicated snapshots container. 
- Removes the issue with WP Snapshots running on PHP 8+ codebases
- Makes Snapshots more extensible


Closes #332 #316 #305

### How to test the Change
- Clone the repo and checkout this branch. 
- Open the directory in a terminal
- Build the codebase 
  - `nvm use`
  - `npm run build`
- Create a new environment and do the following within it
  - Ensure the new WP-CLI command is working
    - `npm run 10updocker wp snapshots search <something>`
  - Ensure the old command is proxying to the new WP-CLI command
    - `npm run 10updocker wpsnapshots search <something>`
    - `npm run 10updocker snapshots search <something>`
- Update an existing environment to ensure the `phpfpm` container can access the required directories (see below)
  - Ensure the new WP-CLI command is working
    - `npm run 10updocker wp snapshots search <something>`
  - Ensure the old command is proxying to the new WP-CLI command
    - `npm run 10updocker wpsnapshots search <something>`
    - `npm run 10updocker snapshots search <something>`

```yaml
  phpfpm:
      image: '10up/wp-php-fpm-dev:7.4-ubuntu'
      # ...
      volumes:
         # ...
          - '~/.wpsnapshots:/home/www-data/.wpsnapshots:cached'
          - '~/.aws:/home/www-data/.aws:cached'
```

### Changelog Entry
> Changed - Snapshots are now using the [Snapshots](https://github.com/10up/snapshots) WP-CLI command


### Credits

Props @tlovett1, @darylldoyle


### Checklist:
